lib/Tools/LinearLayout.cpp

```diff
@@ -16,6 +16,38 @@
 #define DBGS() (llvm::dbgs() << "[" DEBUG_TYPE "]: ")
 #define LDBG(X) LLVM_DEBUG(DBGS() << X << "\n")
 
+#if defined(_MSC_VER) && !defined(__clang__)
+// from https://gist.github.com/pps83/3210a2f980fd02bb2ba2e5a1fc4a2ef0
+#include <intrin.h>
+
+static __forceinline int __builtin_ctz(unsigned x) {
+#if defined(__AVX2__) || defined(__BMI__)
+  return (int)_tzcnt_u32(x);
+#else
```

Comment:
this looks like an unnecessary optimization. I think we can always use `_BitScanForward`